src/Compilers/CSharp/Portable/Binder/UsingStatementBinder.cs

```diff
@@ -245,7 +248,13 @@ bool bindDisposable(bool fromExpression, out MethodArgumentInfo? patternDisposeI
                         awaitableType = originalBinder.Compilation.GetWellKnownType(WellKnownType.System_Threading_Tasks_ValueTask);
                     }
 
-                    return !ReportUseSite(disposableInterface, diagnostics, hasAwait ? awaitKeyword : usingKeyword);
+                    var wasSuccess = !ReportUseSite(disposableInterface, diagnostics, hasAwait ? awaitKeyword : usingKeyword);
+                    if (!wasSuccess)
+                    {
+                        diagnostics.AddRange(patternDiagnostics);
```

Comment:
>diagnostics.AddRange(patternDiagnostics); [](http://example.com/codeflow?start=24&length=41)

Are we testing this code path?

Reply:
Yes, `PatternBasedFails_WithInterfaceImplementation_UseSite`. But the diagnostic from the pattern-based binding is the same, so it's hard to tell by looking at the test...